### PR TITLE
Migrate Done screen to Jetpack Compose

### PR DESCRIPTION
**Pull request by mebarbosa**
## Description
- This migrates the done screen that is displayed after updating the email or password 
- I am proposing to move the done button from the bottom to center because I noticed a problem when we have the mini player active

## Testing Instructions
1. Go to Profile -> Account
2. Smoke tests the change password and change email feature

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [ ] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>State</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt [hunk 1]
```diff
@@ -37,6 +36,20 @@ class DoneViewModel @Inject constructor(
         analyticsTracker.track(AnalyticsEvent.ACCOUNT_UPDATED_DISMISSED)
     }
 
+    sealed class State {
+        data object Empty : State()
+        data class SuccessFullChangedEmail(
+            var detail: String = "",
```
Comment (ashiagr) [inline modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt:42]:
```suggestion
            val detail: String = "",
```

File: modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt [hunk 2]
```diff
@@ -37,6 +36,20 @@ class DoneViewModel @Inject constructor(
...
+            val tittleResourceId: Int = R.string.profile_email_address_changed,
+            val imageResourceId: Int = ic_email_address_changed,
+        ) : State()
+        data class SuccessFullChangedPassword(
+            var detail: String = "",
```
Comment (ashiagr) [inline modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt:47]:
```suggestion
            val detail: String = "",
```

File: modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt [hunk 3]
```diff
@@ -37,6 +36,20 @@ class DoneViewModel @Inject constructor(
         analyticsTracker.track(AnalyticsEvent.ACCOUNT_UPDATED_DISMISSED)
     }
 
+    sealed class State {
+        data object Empty : State()
+        data class SuccessFullChangedEmail(
+            var detail: String = "",
+            val tittleResourceId: Int = R.string.profile_email_address_changed,
```
Comment (ashiagr) [inline modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt:43]: Nitpick: Typo in `tittleResourceId` in both `SuccessFullChangedEmail` and `SuccessFullChangedPassword`.

File: modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt [hunk 4]
```diff
@@ -1,32 +1,31 @@
 package au.com.shiftyjelly.pocketcasts.account.viewmodel
 
-import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import au.com.shiftyjelly.pocketcasts.account.AccountActivity.AccountUpdatedSource
+import au.com.shiftyjelly.pocketcasts.account.R.drawable.ic_email_address_changed
+import au.com.shiftyjelly.pocketcasts.account.R.drawable.ic_password_changed
```
Comment (ashiagr) [inline modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt:6]: Can we use import aliases for resources?
Like: 
`au.com.shiftyjelly.pocketcasts.account.R as AR `
`au.com.shiftyjelly.pocketcasts.localization.R as LR`

File: modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt [hunk 5]
```diff
@@ -37,6 +36,20 @@ class DoneViewModel @Inject constructor(
...
+        data object Empty : State()
+        data class SuccessFullChangedEmail(
+            var detail: String = "",
+            val tittleResourceId: Int = R.string.profile_email_address_changed,
+            val imageResourceId: Int = ic_email_address_changed,
+        ) : State()
+        data class SuccessFullChangedPassword(
+            var detail: String = "",
+            val tittleResourceId: Int = R.string.profile_password_changed,
+            val imageResourceId: Int = ic_password_changed,
```
Comment (ashiagr) [inline modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/DoneViewModel.kt:49]: Can we add annotations like `@StringRes`, `@DrawableRes` wherever applicable?

**Review by ashiagr — APPROVED** [review]
Works well in my testing 👍 

✅ Functionality (`Change Email Address `and `Change Password` both displayed `Done` screen after success)
✅ Scrolling in landscape mode 
✅ Analytics

I left just a few nitpicks. Feel free to merge after these are addressed.

**Comment by mebarbosa** [comment]
I've addressed all comments. Thank you so much @ashiagr !!

**Pull request merged**